posydon/popsyn/synthetic_population.py

```diff
@@ -992,16 +992,38 @@ def __init__(
                 Pwarn(f"{filename} already contains a mass_per_metallicity "
                       "table. Overwriting the table!", "OverwriteWarning")
 
-            simulated_mass = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+            try:
+        # Filter the data based on the condition
+                filtered_data = self.oneline[self.oneline["state_i"] == "initially_single_star"]
+                filtered_data1 = self.oneline[self.oneline["state_i"] != "initially_single_star"]
+    
+        # Calculate the simulated masses
+                simulated_mass_binaries = np.sum(filtered_data1[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                simulated_mass_single = np.sum(filtered_data[["S1_mass_i", "S2_mass_i"]].to_numpy())
+                simulated_mass = np.sum(self.oneline[["S1_mass_i", "S2_mass_i"]].to_numpy())
+    
+            except TypeError as e:
```

Comment:
This exception looks odd to me. What do you try to catch with it and wouldn't it be possible to turn it into a check with two options, hence a `if-else` instead of `try-except`?